pkg/model/file/mixins.go

```diff
@@ -128,3 +130,15 @@ func (m *ResourceMixin) InjectResource(res *resource.Resource) {
 		m.Resource = res
 	}
 }
+
+// ProjectNameMixin provides templates with an injectable project name field.
+type ProjectNameMixin struct {
+	ProjectName string
+}
+
+// InjectProjectName implements HasProjectName.
+func (m *ProjectNameMixin) InjectProjectName(projectName string) {
+	if m.ProjectName == "" {
+		m.ProjectName = strings.ToLower(projectName)
```

Comment:
Do we really want to be calling `strings.ToLower` at this point? This might be confusing to someone seeing `projectName: MyProject` in their `PROJECT` file and then seeing `myproject` in their scaffolded files.

Reply:
Do we have any concept of `PROJECT` file validation anywhere?

Reply:
I'm fine with always lowercasing `projectName`.

Reply:
>Do we have any concept of PROJECT file validation anywhere

No. This file is assumed to be more or less "valid" after `init`.